config/opal_mca.m4

```diff
@@ -145,13 +141,12 @@ AC_DEFUN([OPAL_MCA],[
         IFS="${IFS}$PATH_SEPARATOR,"
         msg=
         for item in $enable_mca_direct; do
-            type="`echo $item | cut -f1 -d-`"
-            comp="`echo $item | cut -f2- -d-`"
+            type="`echo $item | cut -s -f1 -d-`"
+            comp="`echo $item | cut -s -f2- -d-`"
             if test -z $type || test -z $comp ; then
-                AC_MSG_ERROR([*** The enable-mca-direct flag requires a
-*** list of type-component pairs.  Invalid input detected.])
+                AC_MSG_ERROR([enable-mca-direct requires a list of type-component pairs (ex. --enable-mca-direct=pml-ob1,smsc-xpmem)])
             else
-                AS_VAR_SET([AS_TR_SH([DIRECT_$type])], [AS_TR_SH([$comp])])
+                AS_VAR_SET([AS_TR_SH([DIRECT_${type}])], [${comp}])
```

Comment:
I think that the `AS_TR_SH` was just defensive coding (see https://www.gnu.org/software/autoconf/manual/autoconf-2.66/html_node/Common-Shell-Constructs.html).  Hypothetically, neither the `$type` nor the `$comp` should contain characters that can't be in a variable name.

Meaning: if you want to ditch the defensive coding, I guess that's ok.  But it should probably be removed from both, not just one of them, right?

If you really wanted to give a good error message, you could compare `$type` to `AS_TR_SH([$type])`: if they're not the same, then it must be an invalid type name (same for `$comp`).  Extra bonus points will be awarded if you ensure that the framework and component names actually exist in the source tree.

Reply:
Ok, re-added the defensive programming and added support for detecting whether a framework was specified twice or a component does not exits.

Ex:

```
checking if direct-selection component exists for pml framework... configure error: direct component foo requested but not found in cm monitoring ob1 ucx v
```